src/Monolog/Handler/TelegramBotHandler.php

```diff
@@ -83,6 +103,29 @@ protected function send(string $message): void
         $url = self::BOT_API . $this->apiKey . '/SendMessage';
         curl_setopt($ch, CURLOPT_URL, $url);
         curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+
+        if ($this->proxy !== null) {
+            $proxy = parse_url($this->proxy);
+
+            $proxyHost = $proxy['host'];
+
+            if (array_key_exists('port', $proxy)) {
```

Comment:
what about using `isset` instead of `array_key_exists`?
if key `port` in `$proxy` is `null` - you can get some issues with this.